dashboard/app/models/script.rb

```diff
@@ -259,6 +283,20 @@ def self.cache_find_level(level_identifier)
     level
   end
 
+  # Find the LevelConceptDifficulty for the given level from the cache, unless
+  # the level build mode is enabled in which case it is always fetched from the
+  # database.
+  # @param level_id [Integer] the level to fetch the LevelConceptDifficulty of
+  # return [LevelConceptDifficulty | nil] the associated LevelConceptDifficulty
+  def self.cache_find_level_concept_difficulty(level_id)
+     # We intentionally return nil if not found in the cache.
+    level_concept_difficulty_cache[level_id] if self.should_cache?
+
+    # If we aren't using the cache, fetch the LevelConceptDifficulty from the
+    # database.
+    Level.find_by_id(level_id).try(:level_concept_difficulty)
```

Comment:
Again, this could be `LevelConceptDifficulty.find_by_level_id(level_id)`. Doing so would save a DB query, though goes against the paradigm established in the caching case.